🪟 🎉 Adds extra information while deleting source or destination

YatsukBogdan1 opened the pull request:
## What
Closes #19237 

## How
- I added a new `ConfirmationModal` property called `extra` which allows us to add additional information needed by the user.
- I decided to put additional info only while deleting the source or destination since when deleting the connection seems like the user should know what is the result of his actions.
- Also for now there is no styling for extra info content, but maybe we need somehow to highlight this information, it is the design question from this point.

## Screenshots
![Screenshot 2023-01-19 at 16 03 56](https://user-images.githubusercontent.com/25852567/213463951-66c0c2bc-f96d-43cc-8b84-bba805a4956e.png)
![Screenshot 2023-01-19 at 16 03 43](https://user-images.githubusercontent.com/25852567/213463964-8cd1f343-03ea-4216-a0d1-fbb034b3f5e2.png)

edmundito commented:
cc @andyjih for copy changes

dizel852 commented on airbyte-webapp/src/pages/destination/DestinationSettingsPage/DestinationSettingsPage.tsx:
```diff
@@ -57,7 +69,7 @@ export const DestinationSettingsPage: React.FC = () => {
         onSubmit={onSubmitForm}
       />
-      <DeleteBlock type="destination" onDelete={onDelete} />
+      <DeleteBlock extraModal={extraModal} type="destination" onDelete={onDelete} />
```
Hm... Can we replace `extraModal` and `extra` prop name with something more descriptive, like "additionalText", or "additionalInfo"?

edmundito replied:
I suggested `addtionalContent` above

krishnaglick replied:
Could this be `footer`?

edmundito replied:
The modal's footer is where the button go

krishnaglick replied:
+1 to `addtionalContent`

dizel852 commented on airbyte-webapp/src/components/common/ConfirmationModal/ConfirmationModal.tsx:
```diff
@@ -15,12 +15,14 @@ export interface ConfirmationModalProps {
   onClose,
   title,
   text,
+  extra = null,
```
Just curious, why do we set `null` as a default value?

edmundito replied:
This may not be necessary

krishnaglick replied:
I think `undefined` will render out the same as `null`

dizel852 reviewed (changes requested):
Did a review - left some comments.
Also, tests locally - seems like work as expected. 

@edmundito I've not found any figma  design for such a case. Probably would be better to highlight the affected connection names(or at least make them `bold`) WDYT?

edmundito commented on airbyte-webapp/src/locales/en.json:
```diff
@@ -485,6 +485,8 @@
   "tables.connectionDeleteConfirm": "Confirm connection deletion",
   "tables.sourceDeleteModalText": "Deleting a source cannot be undone without a full re-sync. Note that:\n  - All connections with this source will be deleted, including their past logs and configurations.\n  - No existing data in the destination will be altered",
   "tables.destinationDeleteModalText": "Deleting a destination cannot be undone. Note that:\n  - All connections with this destination will be deleted, including their past logs and configurations.\n  - No existing data in the destination will be altered.",
+  "tables.affectedConnectionsOnSourceDeletion": "This connections will be affected if you delete this source:\n",
+  "tables.affectedConnectionsOnDestinationDeletion": "This connections will be affected if you delete this destination:\n",
```
It should be "The following connection will be deleted" instead of "This connections will be affected"

andyjih replied:
Do we have a library for pluralizing? Ideally it would be
1 connection: "The following connection..."
2+ connections: "The following connections..."

edmundito replied:
Yes, the i18n library can pluralize

edmundito commented on airbyte-webapp/src/components/common/ConfirmationModal/ConfirmationModal.tsx:
```diff
@@ -15,12 +15,14 @@ export interface ConfirmationModalProps {
   onSubmit: () => void;
   submitButtonDataId?: string;
   cancelButtonText?: string;
+  extra?: React.ReactNode;
```
`additionalContent` would be a better name.

edmundito commented:
@dizel852 There is no specific design for this. In this case, we expect the developer to use their best judgment.

andyjih reviewed (changes requested):
Copy feedback. Thanks!

YatsukBogdan1 commented:
Next updates were made:
- Added pluralization for connection message
- Changed `extra` to `additionalContent`
- Added `bold` for connection names to make this information recognizable
![Screenshot 2023-01-25 at 14 59 06](https://user-images.githubusercontent.com/25852567/214570157-72a1b434-36b4-470c-aec3-0b18da6f1c49.png)
![Screenshot 2023-01-25 at 14 59 23](https://user-images.githubusercontent.com/25852567/214570174-7fb45800-08d4-4d35-95e5-84b116d3af4a.png)

dizel852 reviewed (approved):
LGTM 👍 

Passing to @edmundito for final review

edmundito commented on airbyte-webapp/src/pages/SourcesPage/pages/SourceItemPage/components/SourceSettings.tsx:
```diff
@@ -54,6 +54,20 @@ const SourceSettings: React.FC<SourceSettingsProps> = ({ currentSource, connecti
+      <p>
+        <FormattedMessage id="tables.affectedConnectionsOnDeletion" values={{ count: connectionsWithSource.length }} />
+        {connectionsWithSource.map((connection) => (
+          <b>- {`${connection.name}\n`}</b>
```
```suggestion
          - <strong>{`${connection.name}\n`}</strong>
```

edmundito commented on airbyte-webapp/src/pages/destination/DestinationSettingsPage/DestinationSettingsPage.tsx:
```diff
@@ -44,6 +44,23 @@ export const DestinationSettingsPage: React.FC = () => {
+          values={{ count: connectionsWithDestination.length }}
+        />
+        {connectionsWithDestination.map((connection) => (
+          <b>- {`${connection.name}\n`}</b>
```
```suggestion
          - <strong>{`${connection.name}\n`}</strong>
```

edmundito reviewed (approved):
Tested locally and looks good, just had a couple of suggestions before merge.

edmundito commented on airbyte-webapp/src/locales/en.json:
```diff
@@ -486,6 +486,7 @@
   "tables.connectionDeleteConfirm": "Confirm connection deletion",
   "tables.sourceDeleteModalText": "Deleting a source cannot be undone without a full re-sync. Note that:\n  - All connections with this source will be deleted, including their past logs and configurations.\n  - No existing data in the destination will be altered",
   "tables.destinationDeleteModalText": "Deleting a destination cannot be undone. Note that:\n  - All connections with this destination will be deleted, including their past logs and configurations.\n  - No existing data in the destination will be altered.",
```
One more thing 😅 : I noticed that while just deleting sources or destinations, the message is now confusing because no connections will be deleted if they don't appear. We can simplify this text to:

```
"Deleting a source cannot be undone without a full re-sync. No existing data in the destination will be altered."

"Deleting a destination cannot be undone. No existing data in the destination will be altered."
```

This way the user only worries about connections being deleted when there are any.
